xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/webjar/Logic.js

```diff
@@ -1114,6 +1114,10 @@ define([
       if (!newEntryValidOperator) {
         newEntry.operator = self.getFilterDefaultOperator(newEntry.property);
       }
+      // If operator has changed, reset value
```

Comment:
This is going to be very annoying when switching the operator doesn't have an impact on the value. For instance the value shouldn't change when switching between ``lowerThan`` and ``greaterThan`` operators of the number filter. Same for the "before" and "after" date operators. So would be better to clear the value only for some operators. E.g. clear the value if "contains" is ether the old or the new operator. Of course, this information is known by the filter not by the logic. So the filter needs to detect that the old/new operator is "contains" and tell logic to clear the value, somehow. WDYT?